internal/util/util.go

```diff
@@ -107,23 +109,35 @@ func CheckExistingBundlesMatchesTemplate(existingBundles *rukpakv1alpha1.BundleL
 }
 
 // CheckDesiredBundleTemplate is responsible for determining whether the existingBundle
-// parameter is semantically equal to the desiredBundle Bundle template.
+// parameter is equal to the desiredBundle Bundle template.
 func CheckDesiredBundleTemplate(existingBundle *rukpakv1alpha1.Bundle, desiredBundle *rukpakv1alpha1.BundleTemplate) bool {
 	// Create a copy of the existingBundle resource and delete the known set of
 	// auto-generated labels that the BI controller injects when comparing the
 	// current and desired set of labels.
-	bundle := existingBundle.DeepCopy()
-	delete(bundle.Labels, CoreOwnerKindKey)
-	delete(bundle.Labels, CoreOwnerNameKey)
-	return equality.Semantic.DeepEqual(bundle.Spec, desiredBundle.Spec) &&
-		equality.Semantic.DeepEqual(existingBundle.Annotations, desiredBundle.Annotations) &&
-		equality.Semantic.DeepEqual(bundle.Labels, desiredBundle.Labels)
+	existingBundle = existingBundle.DeepCopy()
+	delete(existingBundle.Labels, CoreOwnerKindKey)
+	delete(existingBundle.Labels, CoreOwnerNameKey)
+
+	// Split up the bundle name to get the template hash.
+	// Since bundles are immutable, the name of the bundle is guaranteed to conform to what it was upon creation.
+	existingHash := strings.Split(existingBundle.GetName(), "-")[1]
```

Comment:
I believe this logic breaks if the `existingBundle` is named something like `super-cool-bundle` since it is grabbing a different part of the name. Maybe we could consider rehashing the `existingBundle` in this function as a solution?

I added this test case into `TestCheckDesiredBundleTemplate` to check that this is actually a problem:

<details>
<summary>TestCheckDesiredBundleTemplate test case</summary>

```
{
	name: "True/BundleMatchesTemplate",
	args: args{
		existingBundle: &rukpakv1alpha1.Bundle{
			ObjectMeta: metav1.ObjectMeta{
				Name: "stub-123-6f74b48d4",
				Labels: map[string]string{
					"stub-123": "stub-123",
				},
			},
			Spec: sampleSpec,
		},
		desiredBundle: &rukpakv1alpha1.BundleTemplate{
			ObjectMeta: metav1.ObjectMeta{
				Name: "stub-123",
				Labels: map[string]string{
					"stub-123": "stub-123",
				},
			},
			Spec: sampleSpec,
		},
	},
	want: true,
}
```

</details>

Reply:
I don't think rehashing will work because extra stuff gets added to `bundle.metadata` when it's created (resourceVersion, creationTimestamp, generation, etc)

I think rather than parsing the name of the existing bundle, we can store the computed hash in a label on the bundle when we create it and then look it up there when we need to do this comparison. If you look at a replicaset, you'll see `metadata.labels["pod-template-hash"]` with a value equal to the name suffix. We could name ours `bundle-template-hash`.

Reply:
Yes, I realized this was an issue over the weekend -- basically we would have to check the suffix of the name (the last `-`), instead of what was immediately after the first `-`.

As joe pointed out, parsing the name is fairly brittle. I do think that keeping a hash of the template as a label and using that for comparisons is more robust, so I'm leaning towards redoing this towards that implementation instead.

Reply:
Sounds good to me! With whatever we implement it would be great to see a test case (like what I linked) come along with it to verify we're not regressing.